### PR TITLE
Fix toggle publish error

### LINKED ISSUE #69: Fix admin page
The admin page is broken since the toggle published article PR was merged. I'm not sure what the issue is but the page can't be rendered.

### PR DESCRIPTION
**Pull request by salahusama**
Closes #69 
* Fixed toggle errors
* Used editArticle API instead of publish and unpublish
* Updated some libraries to remove risks

Depends on: https://github.com/mai-youth/website-be/pull/26

### Screenshots
image

### CODE REVIEW

File: src/components/admin/ArticleTableRows.js [hunk 1]
```diff
@@ -47,11 +45,15 @@ class ArticleTableRows extends PureComponent {
           <Table.Cell>{formatDate(createdAt)}</Table.Cell>
           <Table.Cell>{formatDate(updatedAt)}</Table.Cell>
           <Table.Cell>
-            <Checkbox toggle defaultChecked={published} onChange={this.togglePublish(id)} />
+            <Checkbox
+              toggle
+              defaultChecked={published === 1}
+              onChange={(e, data) => this.togglePublish(article, data)}
```
Comment (HaashimB) [inline src/components/admin/ArticleTableRows.js]: Not sure what's happening here, has id been split into article and data?

File: package.json [hunk 2]
```diff
@@ -51,12 +49,11 @@
...
-    "cross-env": "^5.2.0",
-    "eslint": "^5.16.0",
+    "cross-env": "^5.2.1",
     "eslint-config-airbnb": "^17.1.0",
-    "eslint-plugin-import": "^2.17.3",
-    "eslint-plugin-jsx-a11y": "^6.2.1",
-    "eslint-plugin-react": "^7.14.2",
+    "eslint-plugin-import": "^2.20.0",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
+    "eslint-plugin-react": "^7.17.0",
```
[the remark this answers is not in the document]
Comment (salahusama) [reply package.json]: Yeah, I think so but I didn't look into it.

I didn't actually update these myself. npm lets you need of security updates and `npm audit fix` or something like that can be used to automatically update them.

But there is a way I think to just specify the major version.

File: src/components/admin/ArticleTableRows.js [hunk 3]
```diff
@@ -4,9 +4,8 @@ import { PropTypes } from 'prop-types';
 import { Table, Button, Checkbox } from 'semantic-ui-react';
 import ArticleFormModal from './ArticleFormModal';
 import ConfirmDialog from './ConfirmDialog';
-import { editArticle, deleteArticle, publishArticle, unpublishArticle } from '../../actions/articles';
+import { editArticle, deleteArticle } from '../../actions/articles';
```
[the remark this answers is not in the document]
Comment (salahusama) [reply src/components/admin/ArticleTableRows.js]: Yes, kind of. It was easier to use `publishArticle` and `unpublishArticle`. I made `editArticle` more general to support the publishes. Needed some changes on the BE as well.

File: src/components/admin/ArticleTableRows.js (hunk 1 above)
Comment (salahusama) [reply src/components/admin/ArticleTableRows.js]: When the value of the checkbox changes _(i.e. its clicked)_, this function is called with the event (`e`) and extra data (`data`). We pass the actual article and this extra data to the `togglePublish`.

`data` contains the new value of the checkbox.

**Pull request merged**